### PR TITLE
feat: validation initial setup and middleware fix

### PR DESCRIPTION
**Pull request by shimulH0**

### CODE REVIEW

File: middleware/auth.ts [hunk 1]
```diff
@@ -1,5 +1,8 @@
 export default defineNuxtRouteMiddleware(() => {
   const { isLoggedIn } = useAuthStore();
-  if (isLoggedIn) return true;
-  return navigateTo("/login");
+  if (!isLoggedIn) {
```
Comment (azmul-ml) [inline middleware/auth.ts:3]: Can't access middleware to check token when refresh/reload the page.

Comment (shimulH0) [reply middleware/auth.ts:3]: bhai don't understand :(

File: pages/login.vue [hunk 2]
```diff
@@ -62,8 +79,12 @@ async function submit() {
           type="password"
           class="form-control"
           placeholder="Enter Password"
+          @blur="v$.password.$touch"
         />
       </div>
+      <div v-for="error of v$.password.$errors" :key="error.$uid" class="input-errors">
+        <div class="error-msg">{{ error.$message }}</div>
+      </div>
       <button type="submit" class="btn btn-primary" :disabled="isLoading" html-type="submit" @click.prevent="submit">
```
Comment (azmul-ml) [inline pages/login.vue:92]: Users can submit the empty forms during create/edit user. It's not expected. have to add validation when trying to submit empty form.

File: pages/register.vue [hunk 3]
```diff
@@ -59,8 +75,12 @@ async function submit() {
           type="password"
           class="form-control"
           placeholder="Enter Password"
+          @blur="v$.password.$touch"
         />
       </div>
+      <div v-for="error of v$.password.$errors" :key="error.$uid" class="input-errors">
+        <div class="error-msg">{{ error.$message }}</div>
+      </div>
       <button html-type="submit" class="btn btn-primary" :disabled="isRegisterLoading" @click.prevent="submit">
```
Comment (azmul-ml) [inline pages/register.vue:88]: Users can submit the empty forms during login/register user.  Have to add validation when trying to submit empty form.

**Review by azmul-ml — COMMENTED** [review]
Please address mentioned comments

**Review by azmul-ml — APPROVED** [review]

**Pull request merged**